### ISSUE TITLE
Python 2.7 Support

### ISSUE THREAD

**Issue by jtdub** [netdevops/hier_config#19]
In reference to #11, our team has been discussing Python 2.7 support, in addition to the Python 3.6 support that the code is currently written to. We certainly appreciate the contributions, but go back and forth on whether it makes sense to walk back code to support 2.7.

Our main concerns with supporting 2.7 are:
* Soon to be deprecated.
  * Python 2.7 will no longer be supported by the community after [January 1, 2020](https://github.com/python/devguide/pull/344) 
  * At least Redhat has announced that their next LTS release of [RHEL will drop support for  Python 2](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/7.5_release_notes/chap-red_hat_enterprise_linux-7.5_release_notes-deprecated_functionality). Others will likely follow soon.
  * By providing 2.7 support, we provide less incentive for others to [update their code to support 3.x ](https://docs.python.org/3/howto/pyporting.html)

* Ability to take advantage of Python 3.x enhancements:
  * Undoing enhancements that are specific to Python 3. It seems counter intuitive to roll back Python 3 specific items to support a version of Python that will soon be deprecated.
  * Ability to use `f` and `r` strings make core more efficient and easier to read.
  * We've been looking at implementing [type hinting](https://docs.python.org/3/library/typing.html). We'd like be able to take advantage of enforcing hints. In Python 2.7, we'd be limited to comment based type hinting, which would limit our ability to enforce hints.

Given this, we'd like to be able to support the community. We'd like to propose some options:

1. Continue supporting Python 3 only support out of the box.
2. Commit to supporting Python 2.7 and Python 3 until Jun 1, 2019. At which time, `hier_config` would likely be converted to Python 3 only. However, Python 2.7 versions would remain on pypi and the last release supporting Python 2.7 would be preserved in a `python2` branch.
3. Create a fork of `hier_config` to specifically support Python 2.7.

**Comment by itdependsnetworks** [comment]
The reality is the infrastructure we work on is what the client says it is. Knowing enterprises, I would not expect it broad py3 support until after the next rhel release.  I would hate to see forking so early in the process.

I think we can build a mutually beneficial relationship, e.g. we have already identified 2 issues, and have plans to extend the library. I appreciate the pain of it all, but it comes down to fstrings being the only thing you mentioned that is not supported in 2.7 (raw string is fine for 2.x, I just removed it, thinking it was another issue.) Additionally, the library scope is much broader if it does support 2.7. 

You know what my vote would be :)
